### PR TITLE
dbeaver/pro#2368 Use monitor from recognition context for query validation

### PR DESCRIPTION
**Pull request by E1izabeth**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/context/SQLQueryCombinedContext.java [hunk 1]
```diff
@@ -43,14 +44,14 @@ public SQLQueryRowsSourceModel findRealSource(@NotNull DBSEntity table) {
 
     @NotNull
     @Override
-    public DBSEntity findRealTable(@NotNull List<String> tableName) {
-        return anyOfTwo(parent.findRealTable(tableName), otherParent.findRealTable(tableName)); // TODO consider ambiguity
+    public DBSEntity findRealTable(List<String> tableName, DBRProgressMonitor monitor) {
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/context/SQLQueryCombinedContext.java]: No @NotNull annotation anymore for tableName?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/context/SQLQueryCombinedContext.java]: thank you

**Review by LonwoLonwo — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLQueryModelRecognizer.java [hunk 2]
```diff
@@ -559,7 +572,7 @@ public void traverseObjs(Object obj) {
     }
 
     @Nullable
-    public SQLQuerySelectionModel recognizeQuery(@NotNull String text) {
+    public SQLQuerySelectionModel recognizeQuery(@NotNull String text, IProgressMonitor monitor) {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLQueryModelRecognizer.java]: I'd use `DBRProgressMonitor` everywhere as possible. When calling such a method, convert the monitor.

I also want to note we mostly pass monitors as the very first parameter of a method. Additionally, nullability annotations would be nice to have here.

This applies to most changes in this PR

**Review by ShadelessFox — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**